plugin/src/main/kotlin/net/octyl/ts2kt/gradle/repository/npm/NpmClientRepository.kt

```diff
@@ -141,15 +157,28 @@ class NpmClientRepository(private val project: Project) : ClientRepository {
 
         // copy to temporary file first
         val temporaryFile = downloadTarget.resolveSibling(".dl.${downloadTarget.name}")
-        val fileChannel = Files.newByteChannel(temporaryFile.toPath(),
-                StandardOpenOption.WRITE,
-                StandardOpenOption.CREATE,
-                StandardOpenOption.TRUNCATE_EXISTING)
+        // hold on to channel to close later
+        val writeChannel = temporaryFile.writeChannel()
         // default arguments don't work here due to KT-24461
-        call.response.content.copyTo(fileChannel)
+        call.response.content.copyTo(writeChannel, Long.MAX_VALUE)
+        // close write channel for safe rename
+        writeChannel.flush()
+        writeChannel.close()
```

Comment:
I'm fairly sure that `close()` implies `flush()`, but perhaps it doesn't.

Additionally, you should probably use `writeChannel.use(() -> R)` that Kotlin provides, as it's the idiomatic way of handling `close()`.

Reply:
Though usually it works out that way, I couldn't find specific documentation saying it did. Better to be safe than sorry?